tests/MongoDB.Driver.Tests/AsyncCursorTests.cs

```diff
@@ -78,17 +82,94 @@ public void KillCursor_should_actually_work()
                 cursor = collection.FindSync("{}");
                 cursor.MoveNext();
 
-                var cursorId = ((AsyncCursor<BsonDocument>) cursor)._cursorId();
+                var cursorId = ((AsyncCursor<BsonDocument>)cursor)._cursorId();
                 cursorId.Should().NotBe(0);
                 cursor.Dispose();
 
                 var desiredResult = BsonDocument.Parse($"{{ \"cursorsKilled\" : [{cursorId}], \"cursorsNotFound\" : [], " +
                     $"\"cursorsAlive\" : [], \"cursorsUnknown\" : [], \"ok\" : 1.0 }}");
-                var result = ((CommandSucceededEvent) eventCapturer.Events[0]).Reply;
+                var result = ((CommandSucceededEvent)eventCapturer.Events[0]).Reply;
                 result.IsSameOrEqualTo(desiredResult);
             }
         }
 
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Tailable_cursor_should_be_able_to_be_cancelled_from_a_different_thread_with_expected_result(
```

Comment:
I like the idea of this test a lot.

But I don't think we need an `async` parameter and I don't think we need an `eventCapturer`.

The intent of the test is to prove that we can cancel out of a blocking call to `enumerator.MoveNext()`.

I think this test can be simplified to:

    [SkippableFact]
    public void Tailable_cursor_should_be_able_to_be_cancelled_from_a_different_thread_with_expected_result()
    {
        RequireServer.Check().Supports(Feature.TailableCursor);

        string testCollectionName = "test";
        string testDatabaseName = "test";

        var client = DriverTestConfiguration.Client;
        var database = client.GetDatabase(testDatabaseName);
        var collection = database.GetCollection<BsonDocument>(testCollectionName);

        DropCollection(client, testDatabaseName, testCollectionName);
        var createCollectionOptions = new CreateCollectionOptions()
        {
            Capped = true,
            MaxSize = 1000
        };
        database.CreateCollection(testCollectionName, createCollectionOptions);
        collection.InsertOne(new BsonDocument()); // tailable cursors don't work on an empty collection

        using (var cancellationTokenSource = new CancellationTokenSource())
        {
            var findOptions = new FindOptions<BsonDocument>()
            {
                BatchSize = 1,
                CursorType = CursorType.TailableAwait
            };
            var cursor = collection.FindSync(FilterDefinition<BsonDocument>.Empty, findOptions);
            var enumerator = cursor.ToEnumerable(cancellationTokenSource.Token).GetEnumerator();

            var semaphore = new SemaphoreSlim(0);
            var thread = new Thread(() =>
            {
                semaphore.Wait();
                cancellationTokenSource.Cancel();
            });
            thread.Start();

            var exception = Record.Exception((Action)(() =>
            {
                while (true)
                {
                    _ = enumerator.MoveNext();
                    semaphore.Release(1);
                }
            }));

            exception.Should().BeOfType<OperationCanceledException>();
        }
    }